### PR TITLE
#255 uwp app crashes

### LINKED ISSUE #255: ⛔ UWP App crashes without a debugger
**Describe the bug**
When running the UWP project without a debugger, the application crashes.

**To Reproduce**
1. Run the UWP app without a debugger (<kbd>Ctrl</kbd> + <kbd>F5</kbd>)
2. The application crashes

**Expected behavior**
The application should be able to run without a debugger.

**Screenshots**
Screenshots are not applicable.

**Desktop (please complete the following information):**
 - OS: Windows 10 (1909, tested on 1903 too, same resultú

**Additional context**
No additional context.

### PR DESCRIPTION
**Pull request by 1n5an1ty**
## Related Issue

Fixes #255 

## Changes

Remove unused reference
Fix xaml element name field (troublesome for some win builds)
Add application unhandled exception handler and restart logic
Update Package.appxmanifest
Fix private var to match project naming schema
Add ILogWriter ref to UI and try/catch around clipboard logic
Fix DI package ref and update all nuget packages to latest stable

## Expected Feedback
Please test and provide any desired feedback. General 5an1ty checks are in order.

### CODE REVIEW

**Comment by 1n5an1ty** [comment]
This one should most likely be merged first among my outstanding PR's once approved/reviewed. If any conflicts arise, which I do expect some, I will resolve in my other PR's. 

Thank you all for your time and energy with these reviews.

File: src/Miunie.WindowsApp/App.xaml.cs [hunk 1]
```diff
@@ -31,8 +35,41 @@ sealed partial class App : Application
...
+
+            var logger = SimpleIoc.Default.GetInstance<ILogWriter>();
+            logger.LogError($"{e.Message}");
+
+            var confirmRestart = new ContentDialog
+            {
+                Title = "Opps! Wasn't expecting that...",
+                Content = "Miunie encountered an error that she was unprepared for. Would you like to reset Miunie with a clean restart?",
+                PrimaryButtonText = "Yes, restart Miunie",
+                CloseButtonText = "Try my luck and continue"
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/App.xaml.cs]: While I always love the creative messages you come up with, this one is more like Windows speaking. 😅 

For that reason, I'd suggest we stick with [Fluent design writing style guidelines](https://docs.microsoft.com/en-us/windows/uwp/design/style/writing-style#error-messages) for errors.

Especially the "Be crisp and clear by eliminating extraneous information." part.

So just a simple "An unexpected error occurred, would you like to restart Miunie, or ignore this problem?"

Or at least something along those lines. 😊 
Sorry to nit-pick like this, I just feel like there is a time and place for style. 😅

File: src/Miunie.WindowsApp/App.xaml.cs [hunk 2]
```diff
@@ -31,8 +35,41 @@ sealed partial class App : Application
...
+            };
+
+            var confirmResult = await confirmRestart.ShowAsync();
+
+            if (confirmResult == ContentDialogResult.Primary) {
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/App.xaml.cs]: _(optional)_ We could invert this if into a return statement to reduce nesting.

File: src/Miunie.WindowsApp/App.xaml.cs [hunk 3]
```diff
@@ -31,8 +35,41 @@ sealed partial class App : Application
...
+
+                var restartResult = await CoreApplication.RequestRestartAsync($"Restart Failure: {e.Message}");
+
+                if (restartResult == AppRestartFailureReason.NotInForeground ||
+                    restartResult == AppRestartFailureReason.RestartPending ||
+                    restartResult == AppRestartFailureReason.Other)
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/App.xaml.cs]: Is there a reason we're not considering the `AppRestartFailureReason.InvalidUser` case?

I haven't looked at what the `RequestRestartAsync` call returns when it goes through, but maybe we could check for that? 🤔 

Or we could hide this boolean expression behind a call at the very least.

Any thoughts on that?

File: src/Miunie.WindowsApp/ViewModels/SettingsPageViewModel.cs [hunk 4]
```diff
@@ -14,7 +14,7 @@ namespace Miunie.WindowsApp.ViewModels
 {
     public class SettingsPageViewModel : ViewModelBase
     {
-        private const string DefaultAvatarUrl = "../Assets/miunie-scarf-transparent.png";
+        private const string _defaultAvatarUrl = "../Assets/miunie-scarf-transparent.png";
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/ViewModels/SettingsPageViewModel.cs]: Our style for private **constants** actually is [Pascal case](http://wiki.c2.com/?PascalCase) so the previous was correct. 😊

File: src/Miunie.WindowsApp/ViewModels/StatusPageViewModel.cs [hunk 5]
```diff
@@ -106,28 +109,36 @@ private async void CheckForTokenInClipboard()
...
-
-            var result = await clipboardTokenDialog.ShowAsync();
-
-            if (result == ContentDialogResult.Primary) { 
-                _tokenManager.ApplyToken(possibleToken, _miunie);
-                RaisePropertyChanged(nameof(SettingsButtonIsVisable));
-                RaisePropertyChanged(nameof(ActionCommand));
+                _logWriter.LogError($"Unable to query clipboard: {ex.Message}");
             }
         }
```
Comment (petrspelos) [inline src/Miunie.WindowsApp/ViewModels/StatusPageViewModel.cs]: _(optional)_
I feel like this try/catch statement is getting a bit complex.
I'd suggest extracting it into `TryApplyTokenFromClipboard()` at the very least. 😊

**Pull request merged**